### ISSUE TITLE
Instructions on using TinyTeX

### ISSUE THREAD

**Issue by yihui** [ulyngs/oxforddown#20]
Following #4, I tried to build this project with TinyTeX, and below is what I found, which could be used to update the LaTeX requirements in readme: https://github.com/ulyngs/oxforddown#requirements

First install the base TinyTeX:

```r
remotes::install_github('yihui/tinytex')
tinytex::install_tinytex()
p1 = tinytex::tl_pkgs()
```

Then render the book, and let **tinytex** automatically install missing LaTeX packages (this will take a while for the first time):

```r
bookdown::render_book('index.Rmd')
p2 = tinytex::tl_pkgs()
```

Check which new packages were required:

```r
p3 = setdiff(p2, p1) 
```

So the instructions on using TinyTeX can tell users to install these additional packages (so normal users won't need to do the above steps by themselves but simply run the code below):

```r
p3 = c(
  "appendix", "babel-english", "babel-greek", "babel-latin", 
  "biber", "biblatex", "caption", "cbfonts-fd", "colortbl", "csquotes", 
  "enumitem", "environ", "eso-pic", "fancyhdr", "greek-fontenc", 
  "grfext", "hyphen-greek", "hyphen-latin", "lineno", "logreq", 
  "makecell", "microtype", "minitoc", "multirow", "notoccite", 
  "oberdiek", "pdflscape", "pdfpages", "quotchap", "soul", "tabu", 
  "threeparttable", "threeparttablex", "titlesec", "tocbibind", 
  "trimspaces", "ulem", "units", "utopia", "varwidth", "wrapfig"
)
tinytex::tlmgr_install(p3)
```

Then this 280Mb TinyTeX should match the 4Gb MiKTeX and produce the expected PDF output.

**Comment by ulyngs** [comment]
This is fantastic, will update the readme! Thank you so much for looking into this, Yihui! 😄

**Comment by yihui** [comment]
> TinyTeX will install these automatically anyway, no?

Yes, but it will take a long time for the first-time users, because TinyTeX will try to install packages one by one only after failed compilations. I don't remember how many attempts it had to make to figure out all missing packages (I remember it took at least five minutes or perhaps ten minutes); `p3` contains 41 packages, so this project may have to be recompiled for ~30 times for the first time.

> Was your point just to clarify what the additional packages are that should be installed by TinyTeX upon the first build?

Exactly.

**Comment by ulyngs** [comment]
I've updated the readme now, thank you Yihui!!

**Comment by yihui** [comment]
Perfect! You are welcome! I'd like to also thank @bttomio for the help with https://github.com/yihui/tinytex/issues/303, which I wouldn't be able to figure out by myself.

**Comment by ulyngs** [comment]
> Perfect! You are welcome! I'd like to also thank @bttomio for the help with [yihui/tinytex#303](https://github.com/yihui/tinytex/issues/303), which I wouldn't be able to figure out by myself.

Does this issue mean that the latex package `psnfss` should be added to the list of additional packages to install with `tinytex::tlmgr_install`? (it's not included in the list above)

**Comment by yihui** [comment]
No, I've included it in the base installation, which is how I fixed yihui/tinytex#303.
